app/components/TransactionHistoryModal.tsx

```diff
@@ -0,0 +1,361 @@
+                                          >
+                                            {transaction.status}
+                                          </span>
+                                        </div>
+                                      </div>
+                                    </div>
+                                    <span className="text-sm text-text-secondary dark:text-white/50">
+                                      {transaction.nativeValue}
+                                    </span>
+                                  </div>
+                                ))}
+                              </div>
+                            </div>
+                          ))}
+                        </div>
+                      </motion.div>
+                    )}
+                  </AnimatePresence>
+                </DialogPanel>
+              </motion.div>
+            </div>
+          </div>
+        </Dialog>
+      )}
+    </AnimatePresence>
+  );
+};
+
+const TransactionDetails = ({ transaction }: { transaction: Transaction }) => {
+  const [isLoading, setIsLoading] = useState(false);
+
+  const handleGetReceipt = () => {
+    setIsLoading(true);
+    setTimeout(() => {
+      setIsLoading(false);
+      console.log("Receipt generated for", transaction.id);
+    }, 1500);
+  };
+
+  return (
+    <div className="space-y-4">
+      <div className="mt-4 flex items-center gap-2">
+        <div className="relative">
+          <Image
+            src={`/logos/${String(transaction.currency)?.toLowerCase()}-logo.svg`}
+            alt={transaction.currency}
+            width={20}
+            height={20}
+            quality={90}
+            className="rounded-full"
+          />
+          <div className="absolute -right-[76%] -top-[1px] z-10 h-fit w-fit rounded-full border-[2px] border-white dark:border-surface-overlay">
+            <Image
+              src={`/logos/${
+                String(transaction.currency)?.toLowerCase() === "usdc"
+                  ? "usdt"
+                  : "usdc"
+              }-logo.svg`}
+              alt={transaction.currency}
+              width={20}
+              height={20}
+              quality={90}
+              className="rounded-full"
+            />
+          </div>
+        </div>
+
+        <span className="ml-2 text-base font-medium dark:text-white">
+          Swapped {transaction.amount} {transaction.currency}
+        </span>
+      </div>
+
+      <p
+        className={classNames(
+          "text-xs",
+          transaction.status === "Completed"
+            ? "text-green-500"
+            : "text-red-500",
+        )}
+      >
+        {transaction.status}
+      </p>
+
+      <div className="space-y-0 pt-4">
+        <div className="w-full border border-dashed border-[#EBEBEF] dark:border-[#FFFFFF1A]" />
+        <div className="flex justify-between py-[14px] dark:border-white/10">
+          <span className="text-sm text-gray-500 dark:text-white/80">
```

Comment:
please check this text color on dark mode with the design, it's inconsistent. it's 50% opacity on the design.